dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -313,6 +316,40 @@ public Response toolsIndexSearch(String query, MultivaluedMap<String, String> qu
         }
     }
 
+    /**
+     * Certain characters need to be escaped in the search term ("include" key in request payload) or the request will fail
+     * See reserved characters here: https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html#regexp-optional-operators
+     * @param query
+     * @return a query with the modified search term
+     */
+    protected static String escapeCharactersInSearchTerm(String query) {
+        if (query != null) {
+            JSONObject json;
+            try {
+                json = new JSONObject(query);
+            } catch (JSONException ex) {
+                LOG.error(SEARCH_QUERY_INVALID_JSON, ex);
+                throw new CustomWebApplicationException(SEARCH_QUERY_INVALID_JSON, HttpStatus.SC_UNSUPPORTED_MEDIA_TYPE);
+            }
+
+            try {
+                String include = getSearchQueryJsonIncludeKey(json);
+                if (include.length() > 0) {
+                    String escapedStr = include.replaceAll("\\.\\*$", "").replaceAll(SEARCH_QUERY_REGEX, "\\\\$1");
```

Comment:
The ui adds a `.*` (match any character) to retrieve words that begin with the search term when the request body includes the "autocomplete" field. That's why the trailing `.*` needs to be removed first before escaping characters, otherwise it will search for a literal `.` and `*`.

We could also not add the `.*` when making the request body and instead add it later in the webservice if that makes more sense logically?

Reply:
Cool, thanks, might be good to add a brief comment to that effect.